src/NuGet.Core/NuGet.Repositories/NuGetv3LocalRepository.cs

```diff
@@ -82,27 +107,10 @@ public IEnumerable<LocalPackageInfo> FindPackagesById(string packageId)
         /// </summary>
         public void ClearCacheForIds(IEnumerable<string> packageIds)
         {
-            lock (_cache)
+            foreach (var packageId in packageIds)
             {
-                foreach (var packageId in packageIds)
-                {
-                    _cache.Remove(packageId);
-                }
-            }
-        }
-
-        private IEnumerable<LocalPackageInfo> GetOrAdd(string packageId, Func<string, List<LocalPackageInfo>> factory)
-        {
-            lock (_cache)
-            {
-                IEnumerable<LocalPackageInfo> results;
-                if (!_cache.TryGetValue(packageId, out results))
-                {
-                    results = factory(packageId);
-                    _cache[packageId] = results;
-                }
-
-                return results;
+                IEnumerable<LocalPackageInfo> packages;
+                _cache.TryRemove(packageId, out packages);
```

Comment:
The remove here invalidates the cache. After a package is installed the clear is called which makes it so that the next call goes to disk and discovers the new packages. The cache is used primarily when writing out the lock file, so along as the cache either contains the new package or has been cleared when that call is made the package will be found.